### PR TITLE
Reimplemented editor cheat, added comments for _Notification

### LINKED ISSUE #974: Add comments to PlayerMicrobeInput _Notification
Add comments to  src/microbe_stage/PlayerMicrobeInput.cs  _Notification(int focus) as to why it is there.

### LINKED ISSUE #1025: Reimplement editor cheat
https://github.com/Revolutionary-Games/Thrive/blob/bf37ff573abcdafe05017db82999c9669449d83c/src/microbe_stage/PlayerMicrobeInput.cs#L69

### PR DESCRIPTION
**Pull request by icedjuro**
Reimplemented the editor cheat with the key 'P'. Added comments explaining what the _Notification method does in PlayerMicrobeInput.cs. Added timer to wait before damaging player from lack of ATP

Closes #1025
Closes #974

### CODE REVIEW

**Comment by icedjuro** [comment]
Closes #964 
Closes #974

**Comment by icedjuro** [comment]
Set wait time to 10 seconds before ATP damage starts taking effect

**Comment by hhyyrylainen** [comment]
ok to test

File: src/microbe_stage/Microbe.cs [hunk 1]
```diff
@@ -1295,15 +1297,18 @@ private void HandleOsmoregulation(float delta)
     /// </summary>
     private void ApplyATPDamage()
     {
-        if (Compounds.GetCompoundAmount("atp") <= 0.0f)
+        if (noATPDamageWait >= Constants.NO_ATP_DAMAGE_TIME)
```
Comment (hhyyrylainen) [inline src/microbe_stage/Microbe.cs:1300]: This is not what I had in mind. The actual cause why the glucose drains immediately and the player runs out of ATP really fast needs to be fixed.

File: src/microbe_stage/PlayerMicrobeInput.cs [hunk 2]
```diff
@@ -24,6 +25,7 @@ public class PlayerMicrobeInput : Node
     public override void _Ready()
     {
         stage = (MicrobeStage)GetParent();
+        microbeHUD = (MicrobeHUD)stage.GetNode("MicrobeHUD");
```
Comment (hhyyrylainen) [inline src/microbe_stage/PlayerMicrobeInput.cs:28]: Why is this like this? There is already `public MicrobeHUD HUD { get; private set; }` in MicrobeStage

File: src/microbe_stage/PlayerMicrobeInput.cs [hunk 3]
```diff
@@ -115,6 +117,8 @@ public override void _Input(InputEvent @event)
 
     public override void _Notification(int focus)
     {
+        // Checks if the game window is not in focus
+        // If so, the player stops moving
```
Comment (hhyyrylainen) [inline src/microbe_stage/PlayerMicrobeInput.cs:121]:
```
If the window goes out of focus we don't receive the key released events, so we need to reset our held down keys to make things work nicely if the player tabs out while pressing a key
```

**Comment by hhyyrylainen** [comment]
Isn't #964 the wrong issue for this to close? That's the one about fixing placing organelle is the editor.

File: src/microbe_stage/PlayerMicrobeInput.cs [hunk 4]
```diff
@@ -66,7 +66,7 @@ public override void _Input(InputEvent @event)
 
         if (@event.IsActionPressed("g_cheat_editor"))
         {
-            // TODO: reimplement editor enter cheat
+            stage.HUD.EditorButtonPressed();
```
Comment (hhyyrylainen) [inline src/microbe_stage/PlayerMicrobeInput.cs:69]: Actually, the cheat used to just enable the editor button, not directly go to the editor. I think I'd like to get that functionality back.

**Comment by icedjuro** [comment]
> Isn't #964 the wrong issue for this to close? That's the one about fixing placing organelle is the editor.
Fixed it

**Pull request merged**